vcd/resource_vcd_network_isolated_v2.go

```diff
@@ -0,0 +1,289 @@
+)
+
+func resourceVcdNetworkIsolatedV2() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdNetworkIsolatedV2Create,
+		ReadContext:   resourceVcdNetworkIsolatedV2Read,
+		UpdateContext: resourceVcdNetworkIsolatedV2Update,
+		DeleteContext: resourceVcdNetworkIsolatedV2Delete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdNetworkIsolatedV2Import,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Network name",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Network description",
+			},
+			"is_shared": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "NSX-V only - share this network with other VDCs in this organization. Default - false",
+			},
+			"gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Gateway IP address",
+			},
+			"prefix_length": &schema.Schema{
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "Network prefix",
+			},
+			"dns1": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "DNS server 1",
+			},
+			"dns2": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "DNS server 1",
+			},
+			"dns_suffix": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "DNS suffix",
+			},
+			"static_ip_pool": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "IP ranges used for static pool allocation in the network",
+				Elem:        networkV2IpRange,
+			},
+		},
+	}
+}
+
+// resourceVcdNetworkIsolatedV2Create
+func resourceVcdNetworkIsolatedV2Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return diag.Errorf("[isolated network v2 create] error retrieving VDC: %s", err)
+	}
+
+	networkType, err := getOpenApiOrgVdcIsolatedNetworkType(d, vdc)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	orgNetwork, err := vdc.CreateOpenApiOrgVdcNetwork(networkType)
+	if err != nil {
+		return diag.Errorf("[isolated network v2 create] error creating Org Vdc isolated network: %s", err)
```

Comment:
OK, this time I won't tell what nit is in this string, try to guess :)
Hint: there are more of it in this PR.

Reply:
Still slipped on it.... Fixed with Final all